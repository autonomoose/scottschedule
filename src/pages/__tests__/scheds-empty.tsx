import React from "react";
import { render, waitFor } from "@testing-library/react";

import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';

import { ManSched, CreateGroup, ModifyGroup } from '../../components/schedgrputil';
import SchedsPage from "../scheds";

/* mock these to keep from testing all the
     sub-components, and generating new snapshots
*/
interface mockCreateGroupProps {
  onComplete: (status: string) => void,
  open: boolean
}
export const mockCreateGroup = (props: mockCreateGroupProps) => {
    return(
      <Box data-testid='newGroup' display={(props.open)?'block': 'none'}>
      <Button onClick={() => props.onComplete('newGroupName')}>Save New</Button>
      <Button onClick={() => props.onComplete('')}>Cancel New</Button>
      </Box>
)};
interface mockModifyGroupProps {
  group: string,
  groupSched: iSchedGroup,
  onComplete: (status: string) => void,
  open: boolean,
}
export const mockModifyGroup = (props: mockModifyGroupProps) => {
    return(
      <Box data-testid='modifyGroup' display={(props.open)?'block': 'none'}>
      <Button onClick={() => props.onComplete(props.group)}>Save Mod</Button>
      <Button onClick={() => props.onComplete('')}>Cancel Mod</Button>
      <List disablePadding dense sx={{marginLeft: '1em'}}>
        <ListItem button key='testsched' onClick={() => props.onComplete('_testgrp!testsched')}>
            testsched - test sched
        </ListItem>
      </List>

      </Box>
)};

interface mockManSchedProps {
  groupSchedName: string, // group!sched or group!_NEW_
  gSchedule: iSchedule,
  onComplete: (status: string) => void,
  open: boolean
}
export const mockManSched = (props: mockManSchedProps) => {
    return(
      <Box data-testid='manSched' display={(props.open)?'block': 'none'}>
      <Button onClick={() => props.onComplete(props.groupSchedName)}>Save Sched</Button>
      <Button onClick={() => props.onComplete('')}>Cancel Sched</Button>
      </Box>
)};

jest.mock('../../components/schedgrputil', () => ({
    ...jest.requireActual('../../components/schedgrputil'),
    fetchSchedGroupsDB: jest.fn(() => Promise.resolve({})),
    ManSched: jest.fn(),
    CreateGroup: jest.fn(),
    ModifyGroup: jest.fn(),
}));

beforeAll(() => {
  (CreateGroup as jest.Mock).mockImplementation(mockCreateGroup);
  (ModifyGroup as jest.Mock).mockImplementation(mockModifyGroup);
  (ManSched as jest.Mock).mockImplementation(mockManSched);
});

const mockEnqueue = jest.fn();
jest.mock('notistack', () => ({
  ...jest.requireActual('notistack'),
  useSnackbar: () => {
    return {
      enqueueSnackbar: mockEnqueue
    };
  }
}));

const mytest = <SchedsPage />;
describe("SchedsPage", () => {
  it("renders no group/scheds correctly", async () => {
    const {asFragment, getByTestId} = render(mytest);
    await waitFor(() => {
        expect(getByTestId('dataBackdrop')).not.toBeVisible();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("handles null for group/scheds fetch", async () => {
    const utils = render(mytest);
    await waitFor(() => {
        expect(utils.getByTestId('dataBackdrop')).not.toBeVisible();
    });
  });
});


