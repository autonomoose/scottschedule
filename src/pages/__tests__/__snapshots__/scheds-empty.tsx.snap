// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchedsPage renders no group/scheds correctly 1`] = `
<div
  data-testid="authenticator"
>
   
  <div
    style="margin: 1rem auto; min-height: 100vh; text-align: center;"
  >
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed mui-fixed css-1bzzxac-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ti0wu3-MuiToolbar-root"
        >
          <button
            aria-label="Open menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1t7w49f-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <a
            href="/home"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <img
                alt="Werner Digital"
                src="../images/wernerdigital-hosted.png"
                width="26"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-4peds9-MuiTypography-root"
              >
                Scottschedule
              </p>
            </div>
          </a>
          <div
            class="MuiBox-root css-1r3535z"
          >
            <a
              href="/usermaint"
              style="margin-right: 1em;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                mockuname
              </p>
            </a>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1vbvugf-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Sign Out
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
      <div
        class="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-1hrb81m-MuiPaper-root-MuiDrawer-paper"
          style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
        >
          <div
            class="MuiBox-root css-1ecnymn"
          >
            <div
              class="MuiBox-root css-xwg089"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                MainÂ Menu
              </h6>
              <button
                aria-label="Close menu"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1drgtl0-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropUpIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m7 14 5-5 5 5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
          />
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <a
              href="/home"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="HomeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Home
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              href="/usermaint"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Account
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              href="/scheds"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Schedules
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              href="/events"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Events
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              href="/help"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Help
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <div
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Logout
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-h8m1q6"
    >
      <main>
        <div
          data-testid="mainPageDisplay"
          style="text-align: left;"
        >
          <div
            class="MuiBox-root css-7kebnq"
          >
            <div
              style="float: left;"
            >
              <nav
                aria-label="Breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1kkz7jp-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      href="/home"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    &gt;
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <span>
                      Schedules
                    </span>
                  </li>
                </ol>
              </nav>
            </div>
            <div>
              <a
                href="/help/scheds"
              >
                Help
              </a>
            </div>
          </div>
          <div
            class="MuiBox-root css-1b1zfbw"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
                style="max-width: 432px; min-width: 394px; flex: 1 1; box-shadow: 5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
              >
                <div
                  class="MuiBox-root css-1kytcjp"
                >
                  Groups (
                  0
                  )
                  <button
                    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1vbvugf-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    New Group
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-1hyfx7x"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
                style="margin-top: 3px; max-width: 350px; min-width: 350px; flex: 1 1; box-shadow: -5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
              >
                <div
                  class="MuiBox-root css-ibnnmb"
                >
                  <form>
                    <div
                      class="MuiBox-root css-h9yv4t"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                      >
                        Add New Group
                      </h6>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-1uqyw5r-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        X
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <label>
                        Name 
                        <input
                          aria-invalid="false"
                          data-testid="nameInput"
                          name="name"
                          size="12"
                          type="text"
                        />
                      </label>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <label>
                         Description
                        <input
                          aria-invalid="false"
                          data-testid="descrInput"
                          name="descr"
                          size="30"
                          type="text"
                        />
                      </label>
                    </div>
                    <div
                      class="MuiBox-root css-10igwnb"
                    >
                      <button
                        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root Mui-disabled  css-1is3qj5-MuiButtonBase-root-MuiButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        Reset
                      </button>
                      <button
                        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root Mui-disabled  css-106f7v8-MuiButtonBase-root-MuiButton-root"
                        disabled=""
                        tabindex="-1"
                        type="submit"
                      >
                        Save
                      </button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-1hyfx7x"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
                style="margin-top: 3px; max-width: 350px; min-width: 350px; flex: 1 1; box-shadow: -5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
              >
                <div
                  class="MuiBox-root css-ibnnmb"
                >
                  <form>
                    <div
                      class="MuiBox-root css-h9yv4t"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                      >
                        Modify Group
                      </h6>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-1uqyw5r-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        X
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="MuiBox-root css-70qvj9"
                    >
                      
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <label>
                        <input
                          aria-invalid="false"
                          data-testid="descrInput"
                          name="descr"
                          size="30"
                          type="text"
                        />
                      </label>
                    </div>
                    <div
                      class="MuiBox-root css-10igwnb"
                    >
                      <button
                        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root Mui-disabled  css-1is3qj5-MuiButtonBase-root-MuiButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        Reset
                      </button>
                      <button
                        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root Mui-disabled  css-106f7v8-MuiButtonBase-root-MuiButton-root"
                        disabled=""
                        tabindex="-1"
                        type="submit"
                      >
                        Save
                      </button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-1hyfx7x"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
                style="margin-top: 3px; max-width: 350px; min-width: 350px; flex: 1 1; box-shadow: -5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
              >
                <div
                  class="MuiBox-root css-ibnnmb"
                >
                  <form>
                    <div
                      class="MuiBox-root css-h9yv4t"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                      >
                        <span>
                          Add Schedule (
                          
                          )
                        </span>
                      </h6>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-1uqyw5r-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="cancel"
                        tabindex="0"
                        type="button"
                      >
                        X
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      class="MuiBox-root css-gg4vpm"
                    >
                      <div
                        class="MuiBox-root css-k008qs"
                      >
                        <label>
                          Name 
                          <input
                            aria-invalid="false"
                            data-testid="nameInput"
                            name="schedName"
                            size="12"
                            type="text"
                          />
                        </label>
                      </div>
                      
                      <div
                        class="MuiBox-root css-0"
                      >
                        <label>
                          Button 
                          <input
                            aria-invalid="false"
                            data-testid="buttonInput"
                            name="buttonName"
                            size="6"
                            type="text"
                          />
                        </label>
                      </div>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <label>
                        <input
                          aria-invalid="false"
                          data-testid="descrInput"
                          name="descr"
                          size="30"
                          type="text"
                        />
                      </label>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <label>
                        Begins 
                        <textarea
                          aria-invalid="false"
                          cols="27"
                          data-testid="beginsInput"
                          name="begins"
                          rows="1"
                        />
                      </label>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <label>
                        Sound 
                        <input
                          aria-invalid="false"
                          data-testid="soundInput"
                          name="sound"
                          size="10"
                          type="text"
                        />
                      </label>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <label>
                        Repeat 
                        <input
                          aria-invalid="false"
                          data-testid="soundRepeatInput"
                          name="soundrepeat"
                          size="2"
                          type="text"
                        />
                      </label>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <label>
                        Warn 
                        <input
                          aria-invalid="false"
                          data-testid="warnInput"
                          name="warn"
                          size="10"
                          type="text"
                        />
                      </label>
                    </div>
                    <div
                      class="MuiBox-root css-10igwnb"
                    >
                      <button
                        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root Mui-disabled  css-1is3qj5-MuiButtonBase-root-MuiButton-root"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        Reset
                      </button>
                      <button
                        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root Mui-disabled  css-106f7v8-MuiButtonBase-root-MuiButton-root"
                        disabled=""
                        tabindex="-1"
                        type="submit"
                      >
                        Save
                      </button>
                    </div>
                  </form>
                  <div
                    class="MuiBox-root css-3uedfb"
                  >
                    <span>
                      Events (
                      0
                      ) 
                    </span>
                    <button
                      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1is3qj5-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      Add Event
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="MuiBox-root css-1hyfx7x"
                  >
                    <div
                      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
                      style="margin-top: 3px; max-width: 350px; min-width: 350px; flex: 1 1; box-shadow: -5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
                    >
                      <div
                        class="MuiBox-root css-ibnnmb"
                      >
                        <form>
                          <div
                            class="MuiBox-root css-h9yv4t"
                          >
                            <h6
                              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                            >
                              Connect Event
                            </h6>
                          </div>
                          <div
                            class="MuiBox-root css-1l4w6pd"
                          >
                            <div
                              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-viou3o-MuiAutocomplete-root"
                              data-testid="taskid"
                            >
                              <div
                                aria-invalid="false"
                                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                              >
                                <label
                                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-rb5gc9-MuiFormLabel-root-MuiInputLabel-root"
                                  data-shrink="false"
                                  for="taskid"
                                  id="taskid-label"
                                >
                                  Event Name
                                </label>
                                <div
                                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1jgdtkb-MuiInputBase-root-MuiOutlinedInput-root"
                                >
                                  <input
                                    aria-autocomplete="list"
                                    aria-expanded="false"
                                    aria-invalid="false"
                                    autocapitalize="none"
                                    autocomplete="off"
                                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1h9uykw-MuiInputBase-input-MuiOutlinedInput-input"
                                    id="taskid"
                                    name="taskid"
                                    role="combobox"
                                    spellcheck="false"
                                    type="text"
                                    value=""
                                  />
                                  <div
                                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                                  >
                                    <button
                                      aria-label="Open"
                                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-1ciwbrp-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                                      tabindex="-1"
                                      title="Open"
                                      type="button"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                        data-testid="ArrowDropDownIcon"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M7 10l5 5 5-5z"
                                        />
                                      </svg>
                                      <span
                                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                      />
                                    </button>
                                  </div>
                                  <fieldset
                                    aria-hidden="true"
                                    class="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
                                  >
                                    <legend
                                      class="css-1ftyaf0"
                                    >
                                      <span>
                                        Event Name
                                      </span>
                                    </legend>
                                  </fieldset>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            class="MuiBox-root css-10igwnb"
                          >
                            <button
                              class="MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-78wnmy-MuiButtonBase-root-MuiButton-root"
                              tabindex="0"
                              type="button"
                            >
                              Cancel
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                            <button
                              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1is3qj5-MuiButtonBase-root-MuiButton-root"
                              tabindex="0"
                              type="button"
                            >
                              Reset
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                            <button
                              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root  css-106f7v8-MuiButtonBase-root-MuiButton-root"
                              tabindex="0"
                              type="submit"
                            >
                              Save
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="MuiBackdrop-root css-1178cxv-MuiBackdrop-root"
            style="opacity: 0; visibility: hidden;"
          >
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
              data-testid="dataBackdrop"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </span>
          </div>
        </div>
      </main>
    </div>
    <footer
      style="padding-top: 40px;"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1bnxa12"
      >
        <a
          href="/home"
        >
          Home
        </a>
        <a
          href="/usermaint"
        >
          Account
        </a>
        <a
          href="/scheds"
        >
          Schedules
        </a>
        <a
          href="/events"
        >
          Events
        </a>
        <a
          href="/help"
        >
          Help
        </a>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-3merus-MuiTypography-root"
      >
        Â© 2021-
        2022
        , Werner Digital Technology Inc
      </span>
    </footer>
  </div>
</div>
`;
